src/com/amazon/ion/impl/IonTypeID.java

```diff
@@ -0,0 +1,117 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.IonType;
+
+/**
+ * Holds pre-computed information about a binary Ion type ID byte.
+ */
+final class IonTypeID {
+
+    private static final int NUMBER_OF_BYTES = 0x100;
+    private static final int BITS_PER_NIBBLE = 4;
+    private static final int LOW_NIBBLE_BITMASK = 0x0F;
+    private static final int NULL_VALUE_NIBBLE = 0xF;
+    private static final int VARIABLE_LENGTH_NIBBLE = 0xE;
+    private static final int NEGATIVE_INT_TYPE_CODE = 0x3;
+    private static final int TYPE_CODE_INVALID = 0xF;
+    private static final int ANNOTATION_WRAPPER_MIN_LENGTH = 0x3;
+    private static final int ANNOTATION_WRAPPER_MAX_LENGTH = 0xE;
+    static final int ORDERED_STRUCT_NIBBLE = 0x1;
+
+    // NOTE: 'annotation wrapper' is not an IonType, but it is simplest to treat it as one for the purposes of this
+    // implementation in order to have a direct mapping from binary type IDs to IonType enum values. IonType.DATAGRAM
+    // does not have a type ID, so we will use it to mean 'annotation wrapper' instead.
+    static final IonType ION_TYPE_ANNOTATION_WRAPPER = IonType.DATAGRAM;
+
+    // Lookup table from type ID to IonType.
+    static final IonType[] ION_TYPES = new IonType[] {
+        IonType.NULL,
+        IonType.BOOL,
+        IonType.INT,
+        IonType.INT,
+        IonType.FLOAT,
+        IonType.DECIMAL,
+        IonType.TIMESTAMP,
+        IonType.SYMBOL,
+        IonType.STRING,
+        IonType.CLOB,
+        IonType.BLOB,
+        IonType.LIST,
+        IonType.SEXP,
+        IonType.STRUCT,
+        ION_TYPE_ANNOTATION_WRAPPER,
+        null // The 0xF type code is illegal in Ion 1.0.
+    };
```

Comment:
Documentation suggestion, include a link to     
```java
// See https://amzn.github.io/ion-docs/docs/binary.html#typed-value-formats
```

Reply:
Done.